runtime_patch.go

```diff
@@ -60,6 +60,99 @@ func updateMagicValue(file *ast.File, magicValue uint32) {
 	}
 }
 
+// updateEntryOffset adds xor encryption for funcInfo.entryoff
+func updateEntryOffset(file *ast.File, entryOffKey uint32) {
+	var nameOffField string
+	entryOffUpdated := false
+
+	extractNameOff := func(node ast.Node) bool {
+		indexExpr, ok := node.(*ast.IndexExpr)
+		if !ok {
+			return true
+		}
+		selExpr, ok := indexExpr.Index.(*ast.SelectorExpr)
+		if !ok {
+			return true
+		}
+		nameOffField = selExpr.Sel.Name
+		return false
+	}
+
+	updateEntryOff := func(node ast.Node) bool {
```

Comment:
it would be helpful if you documented the bit of code that each of these two functions is matching or transforming.

for example, I guess that updateEntryOff is trying to change:

    recv.textAddr(foo.bar)

into... I'm not sure what :) perhaps more helpful if you show an example of an actual call in the runtime package, and how it might look like once transformed.

Reply:
I add a more detailed explanation what and how it works